Add `overwrite` option to skip existing files

jnv opened the pull request:
Hi, we use `s3-batch-upload` to upload files with content hash in their name (e.g. bundled assets from Webpack). I wanted to avoid overwriting files which are already uploaded. Basically if the file exists on the remote end, we are sure that it's the same file – no need to check MD5 or mtime.

Perhaps this feature is too much specific to our use case, but I was going to implement it anyways, so I thought why not propose it to the upstream.

I have added option `overwrite` with default set to `true`. If you set it to `false`, the uploader will first perform `headObject` on the remote end to check if the file exists and if it does, it won't upload it. This could be eventually expanded to e.g. accept a function which determines whether the file should be overwritten or not.

I still need to properly document it, but the code is ready for review, including tests.

ThaNarie commented:
Hi @jnv, thanks for the contribution!

The feature and your change looks good, so no feedback at this point :)

I'll review the documentation once you updated that.

ThaNarie commented on src/lib/cli.ts:
```diff
@@ -103,6 +103,13 @@ yargs
+    default: true,
+    describe: 'Overwrite remote files with the same name, or skip them.',
+    type: 'boolean',
+    nargs: 1,
```
`nargs` is not needed for booleans, only for strings that could require multiple values

jnv commented:
Thanks @ThaNarie! I have added the documentation for `overwrite` option and also added it to CLI.

I had to change the rest of the `uploadFile` method to use `await` because I was getting an error from TS during build: `TS2322: Type '{}' is not assignable to type 'string'.` – likely due to the method returning a Promise _and_ being marked as `async`. The functionality is the same, I hope it's fine.

ThaNarie commented on README.md:
```diff
@@ -47,6 +47,7 @@ Options:
   -acl, --access-control-level  Sets the access control level for uploaded files
                                                                                                   [string] [default: "undefined"]
   -c, --config         The AWS config json path to load S3 credentials with loadFromPath.                       [string]
+  -o, --overwrite      Overwrite remote files with the same name, or skip them.                [boolean] [default: true]
```
Flags that are true by default always feel a big awkward, since you always only use the negative version of it, like `--no-overwrite` in this case.

I think it's worth to explicitly add this to the description, and also as an example more blow (line 60).

jnv replied:
Agreed that having a `overwrite: true` as a default makes this a bit awkward, but I wanted to avoid double negations and keep backward compatibility. I have changed the description to:

> Overwrite remote files with the same name (default behavior), or skip them with --no-overwrite.

And added example.

ThaNarie reviewed (changes requested):
Would be nice if you could look at these 2 comments, I hope you agree!

ThaNarie commented:
This update is available on NPM as `1.4.0` now :)
